File: new-src/Collider.c
#include "Collider.h"
#include "V3.h"
#include "GameObject.h"

Collider *NewColl(GameObject *obj)
{
    Collider *col;
    float x = *obj->scale->x / 2 + *obj->position->x, y = *obj->scale->y / 2 + *obj->position->y, z = *obj->scale->z / 2 + *obj->position->z;
    col->vertices[0] = NewV3(x - *obj->scale->x, y - *obj->scale->y, z);
    col->vertices[1] = NewV3(x, y - *obj->scale->y, z);
    col->vertices[2] = NewV3(x - *obj->scale->x, y - *obj->scale->y, z - *obj->scale->y);
    col->vertices[3] = NewV3(x, y - *obj->scale->y, z - *obj->scale->z);
    col->vertices[4] = NewV3(x - *obj->scale->x, y, z);
    col->vertices[5] = NewV3(x, y, z);
    col->vertices[6] = NewV3(x - *obj->scale->x, y, z - *obj->scale->z);
    col->vertices[7] = NewV3(x, y, z - *obj->scale->z);
    col->xbound = NewV2(col->vertices[4]->x, col->vertices[5]->x);
    col->xbound = NewV2(col->vertices[2]->y, col->vertices[6]->y);
    col->xbound = NewV2(col->vertices[3]->z, col->vertices[1]->z);
    return col;
} 

File: all-src/src/HE/Types/Components/Collider.cs
namespace HE
{
    public struct Collider : Component
    {
        public readonly bool Trigger;
        public Vertice[] vertices = new Vertice[8];
        //0 BTL, 1 BTR, 2 BBL, 3 BBR, 4 TTL, 5 TTR, 6 TBL, 7 TBR

        public Pair xbound, ybound, zbound;

        //0 lx, 1 ly, 2 lz, 3 hx, 4 hy, 5 hz
        internal Collider(Transform transform, bool trigger)
        {
            Trigger = trigger;
            float x = (transform.Scale.x / 2) + transform.Position.x, y = (transform.Scale.y / 2) + transform.Position.y, z = (transform.Scale.z / 2) + transform.Position.z;
            vertices[0] = new Vertice(x - transform.Scale.x, y - transform.Scale.y, z);
            vertices[1] = new Vertice(x, y - transform.Scale.y, z);
            vertices[2] = new Vertice(x - transform.Scale.x, y - transform.Scale.y, z - transform.Scale.z);
            vertices[3] = new Vertice(x, y - transform.Scale.y, z - transform.Scale.z);
            vertices[4] = new Vertice(x - transform.Scale.x, y, z);
            vertices[5] = new Vertice(x, y, z);
            vertices[6] = new Vertice(x - transform.Scale.x, y, z - transform.Scale.z);
            vertices[7] = new Vertice(x, y, z - transform.Scale.z);
            xbound = new Pair(vertices[4].x, vertices[5].x);
            ybound = new Pair(vertices[2].y, vertices[6].y);
            zbound = new Pair(vertices[3].z, vertices[1].z);
        }
    }
}